types/store.go

```diff
@@ -24,6 +25,10 @@ type (
 	Iterator         = types.Iterator
 )
 
+// StoreDecoderRegistry defines each of the modules store decoders. Used for ImportExport
+// simulation.
+type StoreDecoderRegistry map[string]func(cdc *codec.Codec, kvA, kvB cmn.KVPair) string
```

Comment:
Haven't dug too in depth into this PR yet, but after seeing this type and its usage:

- I think it's a grand idea and allows us to keep code DRY and readable
- Doesn't really belong in the module manager (or any core component); rather, can this be used in isolation somehow or maybe even a simulation manager?

Reply:
> Doesn't really belong in the module manager (or any core component); rather, can this be used in isolation somehow or maybe even a simulation manager

I like the concept of a simulation manager as it'd clearly separate the core module logic from the modular funcs for the simulator. @rigelrozanski thoughts?

Reply:
Yeah totally - let's separate out a simulation manager interface! Great idea. With this design I would propose we can have each module's `AppModule` struct fulfill _both_ the simulation `AppModule` interface _as well_ as a new interface `AppSimulationModule` -> this way the `SimulationManager` would take in `AppSimulationModule`'s.

Reply:
Like this abstraction!